record/record_test.go

```diff
@@ -399,13 +399,16 @@ func TestBasicRecover(t *testing.T) {
 }
 
 func TestComplexRecover(t *testing.T) {
```

Comment:
Instead of changing TestComplexRecover from having one consecutive bad block to having two, can we test both of those?

It may be worth refactoring some of this. Go style, by the way, favors table driven tests more than other languages do. See https://github.com/golang/go/wiki/TableDrivenTests and other _test.go files in this repo.